driver_test.go

```diff
@@ -297,8 +211,132 @@ func TestQueryContext(t *testing.T) {
 	}
 
 	// Drop table.
-	err = executeDdlApi(conn, []string{`DROP TABLE TestQueryContext`})
-	if err != nil {
+	if _, err = db.ExecContext(ctx, `DROP TABLE TestQueryContext`); err != nil {
 		t.Error(err)
 	}
 }
+
+// note: IsDdl function does not check validity of statement
+// just that the statement begins with a DDL instruction.
+// Other checking performed by database.
+func TestIsDdl(t *testing.T) {
+
+	tests := []struct {
+		name  string
+		input string
+		want  bool
+	}{
+		{
+			name: "valid create",
+			input: `CREATE TABLE Valid (
+				A   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			want: true,
+		},
+		{
+			name: "leading spaces",
+			input: `    CREATE TABLE Valid (
+				A   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			want: true,
+		},
+		{
+			name: "leading newlines",
+			input: `
+
+
+			CREATE TABLE Valid (
+				A   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			want: true,
+		},
+		{
+			name: "leading tabs",
+			input: `		CREATE TABLE Valid (
+				A   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			want: true,
+		},
+		{
+			name: "leading whitespace, miscelanious",
+			input: `
+							 
+			 CREATE TABLE Valid (
+				A   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			want: true,
+		},
+		{
+			name: "lower case",
+			input: `create table Valid (
+				A   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			want: true,
+		},
+		{
+			name: "mixed case, leading whitespace",
+			input: ` 
+			 cREAte taBLE Valid (
+				A   STRING(1024)
+			)	 PRIMARY KEY (A)`,
+			want: true,
+		},
+		{
+			name:  "insert (not ddl) ",
```

Comment:
nit: remove trailing space from name. Also applies to below cases.

Reply:
removed